### ISSUE TITLE
2.2.3 -> 2.2.4: New URL protocol unsupported (requires https)

### ISSUE THREAD

**Issue by marshfire** [winfinit/mongodb-download#26]
The change in URL to http://downloads.mongodb.org does not work in our test environment, it complains about the not fitting protocol and requests https.

If the URL protocol gets changed to https://downloads.mongodb.org everything works again

**Comment by franky-continu** [comment]
PLEASE +1

**Comment by franky-continu** [comment]
@marshfire I use mockgoose and traced the error all the way back here.

Is there a version of Mockgoose I can use that won't use (eventually) this version with this error?

how can I workaround the issue for now?

**Comment by franky-continu** [comment]
How can I bundle a changed/forked version with the correct HTTPS url in it and then finally end up using it with mockgoose

**Comment by marshfire** [comment]
@franky-continu well, we originally had the same problem with mockgoose, we ended up putting a modified version with the hard coded previous version of mockgoose and mongodb_prebuilt to a private registry, a workaround for you could be to use shrink-wrap and force the previous version of mongodb_download (2.2.3)

**Comment by franky-continu** [comment]
@marshfire right right. Thank you so much, I'll try to do that

**Comment by franky-continu** [comment]
I added a peerDependencies section to my package.json

```json
  "peerDependencies": {
    "mongodb_download": "2.2.3"
  }
  "devDependencies": {
    "mocha": "^4.0.1",
    "mocha-logger": "^1.0.5",
    "mockgoose": "^7.3.3",
    "request": "^2.83.0",
    "should": "^13.1.3",
    "supertest": "^3.0.0"
  }
```

**Comment by franky-continu** [comment]
That didn't work and my srink-wrap file also didn't work with version 2.2.3

**Comment by alexraj** [comment]
You can also change the 
`var http = require('https')`
to
`var http = require('http')
`

The workaround I did is, copy the mongodb-download.js under node_modules/mongodb-download/built, and made the change as patch. In the Jenkins build step, after "npm install" copy this file over. I think this is a simple workaround, until a fixed version is released, which hopefully is not far away.

**Comment by franky-continu** [comment]
I patch it with a Dockerfile COPY statement . It still fails now with an unauthorized 403  error.

Tried both `require('http');` and the DOWNLOAD_URI with `http://`

and `require('https');` and the DOWNLOAD_URI with `https://`

```
Continu PATCHED version of mongodb-download.js
unable to get signature content { StatusCodeError: 403 - "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>104588471ACA6595</RequestId><HostId>Ekya++RcPpAdUOmQ9XSYMUDswK+iQJ8jBI/+dz4HwH83ucM9f3MhtYZy9PYLhkLJO5tVJIvh/+8=</HostId></Error>"
at new StatusCodeError (/usr/src/app/node_modules/request-promise-core/lib/errors.js:32:15)
at Request.plumbing.callback (/usr/src/app/node_modules/request-promise-core/lib/plumbing.js:104:33)
at Request.RP$callback [as _callback] (/usr/src/app/node_modules/request-promise-core/lib/plumbing.js:46:31)
at Request.self.callback (/usr/src/app/node_modules/request/request.js:186:22)
at emitTwo (events.js:126:13)
at Request.emit (events.js:214:7)
at Request.<anonymous> (/usr/src/app/node_modules/request/request.js:1163:10)
at emitOne (events.js:116:13)
at Request.emit (events.js:211:7)
at IncomingMessage.<anonymous> (/usr/src/app/node_modules/request/request.js:1085:12)
at Object.onceWrapper (events.js:313:30)
at emitNone (events.js:111:20)
at IncomingMessage.emit (events.js:208:7)
at endReadableNT (_stream_readable.js:1056:12)
at _combinedTickCallback (internal/process/next_tick.js:138:11)
at process._tickCallback (internal/process/next_tick.js:180:9)
name: 'StatusCodeError',
statusCode: 403,
message: '403 - "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>\\n<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>104588471ACA6595</RequestId><HostId>Ekya++RcPpAdUOmQ9XSYMUDswK+iQJ8jBI/+dz4HwH83ucM9f3MhtYZy9PYLhkLJO5tVJIvh/+8=</HostId></Error>"',
error: '<?xml version="1.0" encoding="UTF-8"?>\n<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>104588471ACA6595</RequestId><HostId>Ekya++RcPpAdUOmQ9XSYMUDswK+iQJ8jBI/+dz4HwH83ucM9f3MhtYZy9PYLhkLJO5tVJIvh/+8=</HostId></Error>',
options:
{ uri: 'https://downloads.mongodb.org/linux/mongodb-linux-x86_64-debian81-3.2.2.tgz.md5',
callback: [Function: RP$callback],
transform: undefined,
```

**Comment by ihgreenman** [comment]
+1

Also note that the patched version must go under the bin directory, changing DOWNLOAD_URI protocol to https did the trick for me.

**Comment by franky-continu** [comment]
For me it was changing 

```javascript

var DOWNLOAD_URI = "https://fastdl.mongodb.org";

```

And also requesting version 3.4.10 instead of 3.2.2

**Comment by kuncevic** [comment]
got same thing here https://gitter.im/nestjs/nestjs?at=5a20df37540c78242d6d6342 that is happening when runnin this one https://github.com/nestjs/nest/tree/master/examples/10-mockgoose with `process.env.NODE_ENV = 'test'` and without having mongodb installed as the project just using `Mockgoose` thing https://github.com/nestjs/nest/blob/master/examples/10-mockgoose/src/modules/database/database.providers.ts#L10 :
```
import * as mongoose from 'mongoose';
import { Mockgoose } from 'mockgoose';

export const databaseProviders = [
  {
    provide: 'DbToken',
    useFactory: async () => {
      (mongoose as any).Promise = global.Promise;

      if (process.env.NODE_ENV === 'test') {

          const mockgoose = new Mockgoose(mongoose);
          mockgoose.helper.setDbVersion('3.4.3');

          mockgoose.prepareStorage()
              .then(async () => {
                  await mongoose.connect('mongodb://example.com/TestingDB', {
                      useMongoClient: true,
                  });
              });

      } else {

          await mongoose.connect('mongodb://localhost/nest', {
              useMongoClient: true,
          });
      }

      return mongoose;
    },
  },
];
```

**Comment by dtzar** [comment]
This issue is now resolved and can be closed

**Comment by LoneRifle** [comment]
Fixed in #27
